lightning-background-processor/src/lib.rs

```diff
@@ -254,19 +258,44 @@ fn update_scorer<'a, S: 'static + Deref<Target = SC> + Send + Sync, SC: 'a + Wri
 	}
 }
 
+#[cfg(feature = "futures")]
+enum CompletableSleeper<T: Future<Output=bool> + core::marker::Unpin> {
+	Sleep(T),
+	Complete(bool),
+}
+#[cfg(feature = "futures")]
+impl<T: Future<Output=bool> + core::marker::Unpin> Future for CompletableSleeper<T> {
+	type Output = bool;
+
+	fn poll(mut self: core::pin::Pin<&mut Self>, cx: &mut core::task::Context) -> core::task::Poll<bool> {
+		match &mut *self {
+			Self::Sleep(f) => <T as Future>::poll(core::pin::Pin::new(f), cx),
+			Self::Complete(v) => core::task::Poll::Ready(*v),
+		}
+	}
+}
+#[cfg(feature = "futures")]
+impl<T: Future<Output=bool> + core::marker::Unpin> FusedFuture for CompletableSleeper<T> {
+	fn is_terminated(&self) -> bool {
+		match self { Self::Sleep(_) => false, Self::Complete(_) => true }
```

Comment:
`is_terminated` should always return true if the future has ever returned `Poll::Ready`, otherwise the inner future might get polled after completion. If the inner `Sleep` future resolves, this will still return `is_terminated = false`.